aibolit/metrics/npath/main.py

```diff
@@ -103,16 +103,15 @@ def _node_npath(self, node: ASTNode) -> int:
         if isinstance(node, list):
             return self._sequence_npath(node)
 
-        if node.node_type == ASTNodeType.IF_STATEMENT:
-            return self._if_npath(node)
-        elif node.node_type == ASTNodeType.SWITCH_STATEMENT:
-            return self._switch_npath(node)
-        elif node.node_type == ASTNodeType.FOR_STATEMENT:
-            return self._for_loop_npath(node)
-        elif node.node_type == ASTNodeType.BINARY_OPERATION:
-            return self._binary_expression_npath(node)
-        else:
-            return self._sequence_npath(node.children)
+        dispatcher = {
+            ASTNodeType.IF_STATEMENT: self._if_npath,
+            ASTNodeType.SWITCH_STATEMENT: self._switch_npath,
+            ASTNodeType.FOR_STATEMENT: self._for_loop_npath,
+            ASTNodeType.WHILE_STATEMENT: self._while_loop_npath,
+            ASTNodeType.BINARY_OPERATION: self._binary_expression_npath,
+        }
+        handler = dispatcher.get(node.node_type, lambda n: self._sequence_npath(n.children))
+        return handler(node)
```

Comment:
Here I had to perform refactoring towards this dictionary dispatch pattern to avoid too many returns.
In particular, the linter was complaining about 7 return statements with a max of 6.
```
aibolit/metrics/npath/main.py:102:4: R0911: Too many return statements (7/6) (too-many-return-statements)
```

We can consider reverting and silencing the warning if the original implementation is considered more readable.

Alternatively, we can simply write resulting value to a variable `result` and then only return `result` once.

Reply:
@AntonProkopyev I extracted a separate function `default_handler`. I hope it helps convey the message better than an anonymous function. Does it look OK?